infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -180,9 +187,36 @@ public void run() {
         } catch (Exception e) {
             log.error("Unhandled exception caught during log replication service discovery. Retry,", e);
         } finally {
-            if (runtime != null) {
-                runtime.shutdown();
-            }
+            runtime.cleanup(CorfuRuntime::shutdown);
+        }
+    }
+
+    private LockConfig createLockConfig(String pluginFilePath) {
```

Comment:
It is cleaner to have a lock config class.

Reply:
I had to modify this method, so it depends on LogReplicationPluginConfig, which is in infrastructure. This will create a circular dependency between the two. I think it's ok to have it here for now because the LockConfig from a file is actually being created only by the CorfuReplicationDiscvoeryService.

Reply:
There are two options to do:
1. Define a new class extends from DiscoveryServiceClass , the new class is used for testing mostly.
2. Pass the instance of discovery service to the LockConfig class. 
We can do it later. Just this class file becomes so big now. It is hard for people to read in future.